index.js

```diff
@@ -1,97 +1,226 @@
+}
+
+function handleLocalError(urlName, responseDate) {
+    console.log('Local error, couldn\'t reach endpoint.');
+
+    if (urlErrorContainer[urlName] === undefined) {
+        urlErrorContainer[urlName] = responseDate;
+    }
+}
+
+function handleDownStatus(urlName, urlValue, responseDate, mentionsWhenDown) {
+    const trailingMentions = buildMentionsForDownStatus(mentionsWhenDown);
+    const message = `:warning: Oh no! It looks like ${urlName} (${urlValue}) is down.\n${trailingMentions}`;
+    console.log(`${getFormattedDate(responseDate)} - ${message}`);
+
+    if (urlContainer[urlName] === undefined) {
+        send("First check of this session... Endpoint didn't respond :confused:");
+        urlContainer[urlName] = UrlStatus.Pending;
+        return;
+    }
+
+    if (urlErrorContainer[urlName] !== undefined) {
+        restoreLocalErrorAndSend(responseDate, urlName);
+    }
+
+    if (urlContainer[urlName] === UrlStatus.Up) {
+        // First fail! Try again, maybe this was temporary.
+        urlContainer[urlName] = UrlStatus.Pending;
+    } else if (urlContainer[urlName] === UrlStatus.Pending) {
+        // Two fails in a row! Send message, the endpoint is down.
+        urlContainer[urlName] = UrlStatus.Down;
+        send(message);
+    }
+}
+
+function buildMentionsForDownStatus(mentions) {
+    if (mentions === undefined) return '';
+
+    return mentions.reduce((total, mention, index) => {
+        if (mention.kind === undefined || mention.id === undefined) return total;
+
+        const leadingText = (index === 0) ? 'cc: ' : ', ';
+        const trailingText = (mention.kind === 'user') ? `<@${mention.id}>` : `<!subteam^${mention.id}>`;
+
+        return `${total}${leadingText}${trailingText}`
+    }, '');
 }
 
 /* Send message to slack incoming webhook */
 function send(text) {
-    webhook.send(text, function (err, res) {
-        if (err) {
-            console.log('Error:', err);
-        } else {
-            console.log('Message sent: ', res);
+    console.log(webhookUrl);
+    request.post(
+        webhookUrl,
+        {
+            json: { text: text }
+        },
+        (err, res) => {
+            if (err) {
+                console.log(`Error: ${err}`)
+            } else {
+                console.log(`Message sent: ${res}`)
+            }
         }
-    });
+    );
+}
+
+function restoreLocalErrorAndSend(currentDate, urlName) {
+    const localErrorSince = urlErrorContainer[urlName];
+
+    if (localErrorSince === undefined) return;
+
+    const msSpentInLocalError = currentDate.getTime() - localErrorSince.getTime();
+    const hasBeenMoreThanOneMinute = msSpentInLocalError > 60000
+
+    if (hasBeenMoreThanOneMinute) {
+        const hoursSpent = Math.trunc(msSpentInLocalError / (1000 * 3600));
+        const minSpent = Math.trunc((msSpentInLocalError / (1000 * 60)) - (hoursSpent * 60));
+        const secondsSpent = Math.trunc((msSpentInLocalError / 1000) - (minSpent * 60));
```

Comment:
nitpick: you'd better convert this to seconds (/1000), then use div + divmod (%) to split it into hours, minutes and seconds.

Reply:
or since you're using momentjs now, you can use `diff` to get a duration and then use that https://momentjscom.readthedocs.io/en/latest/moment/08-durations/14-diffing/

Reply:
https://momentjscom.readthedocs.io/en/latest/moment/10-plugins/18-duration-format/ may be helpful